src/test/lib/ApiTest.cpp

```diff
@@ -5321,3 +5321,371 @@ QuicTestStorage()
+#define MSQUIC_APP_PARAMETERS_PATH      "System\\CurrentControlSet\\Services\\MsQuic\\Parameters\\Apps\\StorageTest"
+    RegDeleteKeyValueA(
+        HKEY_LOCAL_MACHINE,
+        MSQUIC_GLOBAL_PARAMETERS_PATH,
+        QUIC_SETTING_ACCEPTABLE_VERSIONS);
+    RegDeleteKeyValueA(
+        HKEY_LOCAL_MACHINE,
+        MSQUIC_GLOBAL_PARAMETERS_PATH,
+        QUIC_SETTING_OFFERED_VERSIONS);
+    RegDeleteKeyValueA(
+        HKEY_LOCAL_MACHINE,
+        MSQUIC_GLOBAL_PARAMETERS_PATH,
+        QUIC_SETTING_FULLY_DEPLOYED_VERSIONS);
+    RegDeleteKeyA(
+        HKEY_LOCAL_MACHINE,
+        MSQUIC_APP_PARAMETERS_PATH);
+    HKEY Key;
+    RegCreateKeyA(
+        HKEY_LOCAL_MACHINE,
+        MSQUIC_APP_PARAMETERS_PATH,
+        &Key);
+    RegCloseKey(Key);
+#else
+    TEST_FAILURE("Storage tests not supported on this platform");
+#endif
+
+    MsQuicVersionSettings Settings{};
+
+    //
+    // Global settings
+    //
+
+    TEST_QUIC_SUCCEEDED(Settings.GetGlobal());
+    TEST_EQUAL(Settings.AcceptableVersionsLength, 0);
+    TEST_EQUAL(Settings.OfferedVersionsLength, 0);
+    TEST_EQUAL(Settings.FullyDeployedVersionsLength, 0);
+    TEST_EQUAL(Settings.AcceptableVersions, nullptr);
+    TEST_EQUAL(Settings.OfferedVersions, nullptr);
+    TEST_EQUAL(Settings.FullyDeployedVersions, nullptr);
+
+#ifdef _KERNEL_MODE
+    TEST_QUIC_SUCCEEDED(
+        ZwSetValueKey(
+            GlobalKey,
+            (PUNICODE_STRING)&AcceptableVersionsValueName,
+            0,
+            REG_BINARY,
+            (PVOID)&VersionList,
+            sizeof(VersionList)));
+    TEST_QUIC_SUCCEEDED(
+        ZwSetValueKey(
+            GlobalKey,
+            (PUNICODE_STRING)&OfferedVersionsValueName,
+            0,
+            REG_BINARY,
+            (PVOID)&VersionList,
+            sizeof(VersionList)));
+    TEST_QUIC_SUCCEEDED(
+        ZwSetValueKey(
+            GlobalKey,
+            (PUNICODE_STRING)&FullyDeployedVersionsValueName,
+            0,
+            REG_BINARY,
+            (PVOID)&VersionList,
+            sizeof(VersionList)));
+#elif _WIN32
+    TEST_EQUAL(
+        NO_ERROR,
+        RegSetKeyValueA(
+            HKEY_LOCAL_MACHINE,
+            MSQUIC_GLOBAL_PARAMETERS_PATH,
+            QUIC_SETTING_ACCEPTABLE_VERSIONS,
+            REG_BINARY,
+            &VersionList,
+            sizeof(VersionList)));
+    TEST_EQUAL(
+        NO_ERROR,
+        RegSetKeyValueA(
+            HKEY_LOCAL_MACHINE,
+            MSQUIC_GLOBAL_PARAMETERS_PATH,
+            QUIC_SETTING_OFFERED_VERSIONS,
+            REG_BINARY,
+            &VersionList,
+            sizeof(VersionList)));
+    TEST_EQUAL(
+        NO_ERROR,
+        RegSetKeyValueA(
+            HKEY_LOCAL_MACHINE,
+            MSQUIC_GLOBAL_PARAMETERS_PATH,
+            QUIC_SETTING_FULLY_DEPLOYED_VERSIONS,
+            REG_BINARY,
+            &VersionList,
+            sizeof(VersionList)));
+#else
+    TEST_FAILURE("Storage tests not supported on this platform");
```

Comment:
No need to continue doing TEST_FAILURE, since you have one above. Same for others below.